### ISSUE TITLE
Autocommand error when opening python file

### ISSUE THREAD

**Issue by pta2002** [nvim-treesitter/nvim-treesitter#554]
**Describe the bug**
Whenever I open a python file, I get this error:

```
Error detected while processing FileType Autocommands for "*":
E5108: Error executing lua ...al/share/nvim/runtime/lua/vim/treesitter/highlighter.lua:76: attempt to index a nil value
```
It still works fine if I press ESC, so this is more of an annoyance than anything.

**To Reproduce**
1. Install nvim-treesitter
2. Have this lua code on `init.vim`
```lua
require'nvim-treesitter.configs'.setup {
    ensure_installed = "all",
    highlight = {
        enable = true,
    }
}
```
3. Set up treesitter with `:TSInstall`
4. Open a python file

**Expected behavior**
No error should happen when opening a python file.

**Output of `:checkhealth nvim_treesitter`**

<details>
<code>

health#nvim_treesitter#check
========================================================================
## Installation
  - OK: `git` executable found.
  - OK: `cc` executable found.

## elm parser healthcheck
  - OK: elm parser found.
  - WARNING: No `highlights.scm` query found for elm
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `locals.scm` query found for elm
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `textobjects.scm` query found for elm
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `folds.scm` query found for elm
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

## c parser healthcheck
  - OK: c parser found.
  - OK: `highlights.scm` found.
  - OK: `locals.scm` found.
  - OK: `textobjects.scm` found.
  - OK: `folds.scm` found.

## java parser healthcheck
  - OK: java parser found.
  - OK: `highlights.scm` found.
  - OK: `locals.scm` found.
  - WARNING: No `textobjects.scm` query found for java
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `folds.scm` query found for java
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

## python parser healthcheck
  - OK: python parser found.
  - OK: `highlights.scm` found.
  - ERROR: Error in `locals.scm` query found for python
    /usr/local/share/nvim/runtime/lua/vim/treesitter/query.lua:20: query: invalid node type at position 1421
    - ADVICE:
      - Try `:TSUpdate python` (queries might have been adapted for a upstream parser change)
      - Try to find the syntax error/invalid node type in above file.
  - OK: `textobjects.scm` found.
  - OK: `folds.scm` found.

## dart parser healthcheck
  - OK: dart parser found.
  - OK: `highlights.scm` found.
  - OK: `locals.scm` found.
  - WARNING: No `textobjects.scm` query found for dart
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `folds.scm` query found for dart
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

## lua parser healthcheck
  - OK: lua parser found.
  - OK: `highlights.scm` found.
  - OK: `locals.scm` found.
  - OK: `textobjects.scm` found.
  - OK: `folds.scm` found.

## ocaml parser healthcheck
  - OK: ocaml parser found.
  - OK: `highlights.scm` found.
  - OK: `locals.scm` found.
  - WARNING: No `textobjects.scm` query found for ocaml
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - OK: `folds.scm` found.

## go parser healthcheck
  - OK: go parser found.
  - OK: `highlights.scm` found.
  - OK: `locals.scm` found.
  - OK: `textobjects.scm` found.
  - WARNING: No `folds.scm` query found for go
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

## nix parser healthcheck
  - OK: nix parser found.
  - WARNING: No `highlights.scm` query found for nix
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `locals.scm` query found for nix
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `textobjects.scm` query found for nix
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `folds.scm` query found for nix
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

## yaml parser healthcheck
  - OK: yaml parser found.
  - WARNING: No `highlights.scm` query found for yaml
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `locals.scm` query found for yaml
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `textobjects.scm` query found for yaml
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `folds.scm` query found for yaml
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

## json parser healthcheck
  - OK: json parser found.
  - OK: `highlights.scm` found.
  - WARNING: No `locals.scm` query found for json
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `textobjects.scm` query found for json
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `folds.scm` query found for json
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

## jsdoc parser healthcheck
  - OK: jsdoc parser found.
  - WARNING: No `highlights.scm` query found for jsdoc
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `locals.scm` query found for jsdoc
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `textobjects.scm` query found for jsdoc
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `folds.scm` query found for jsdoc
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

## markdown parser healthcheck
  - OK: markdown parser found.
  - WARNING: No `highlights.scm` query found for markdown
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `locals.scm` query found for markdown
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `textobjects.scm` query found for markdown
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `folds.scm` query found for markdown
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

## julia parser healthcheck
  - OK: julia parser found.
  - WARNING: No `highlights.scm` query found for julia
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `locals.scm` query found for julia
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `textobjects.scm` query found for julia
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `folds.scm` query found for julia
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

## html parser healthcheck
  - OK: html parser found.
  - OK: `highlights.scm` found.
  - OK: `locals.scm` found.
  - WARNING: No `textobjects.scm` query found for html
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `folds.scm` query found for html
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

## typescript parser healthcheck
  - OK: typescript parser found.
  - OK: `highlights.scm` found.
  - OK: `locals.scm` found.
  - OK: `textobjects.scm` found.
  - OK: `folds.scm` found.

## fennel parser healthcheck
  - OK: fennel parser found.
  - OK: `highlights.scm` found.
  - OK: `locals.scm` found.
  - WARNING: No `textobjects.scm` query found for fennel
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `folds.scm` query found for fennel
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

## swift parser healthcheck
  - OK: swift parser found.
  - WARNING: No `highlights.scm` query found for swift
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `locals.scm` query found for swift
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `textobjects.scm` query found for swift
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `folds.scm` query found for swift
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

## query parser healthcheck
  - OK: query parser found.
  - OK: `highlights.scm` found.
  - OK: `locals.scm` found.
  - WARNING: No `textobjects.scm` query found for query
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `folds.scm` query found for query
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

## cpp parser healthcheck
  - OK: cpp parser found.
  - OK: `highlights.scm` found.
  - OK: `locals.scm` found.
  - OK: `textobjects.scm` found.
  - OK: `folds.scm` found.

## ruby parser healthcheck
  - OK: ruby parser found.
  - OK: `highlights.scm` found.
  - OK: `locals.scm` found.
  - WARNING: No `textobjects.scm` query found for ruby
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `folds.scm` query found for ruby
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

## vue parser healthcheck
  - OK: vue parser found.
  - WARNING: No `highlights.scm` query found for vue
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `locals.scm` query found for vue
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `textobjects.scm` query found for vue
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `folds.scm` query found for vue
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

## regex parser healthcheck
  - OK: regex parser found.
  - OK: `highlights.scm` found.
  - WARNING: No `locals.scm` query found for regex
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `textobjects.scm` query found for regex
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `folds.scm` query found for regex
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

## scala parser healthcheck
  - OK: scala parser found.
  - WARNING: No `highlights.scm` query found for scala
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `locals.scm` query found for scala
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `textobjects.scm` query found for scala
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `folds.scm` query found for scala
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

## toml parser healthcheck
  - OK: toml parser found.
  - OK: `highlights.scm` found.
  - OK: `locals.scm` found.
  - WARNING: No `textobjects.scm` query found for toml
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `folds.scm` query found for toml
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

## rust parser healthcheck
  - OK: rust parser found.
  - OK: `highlights.scm` found.
  - OK: `locals.scm` found.
  - OK: `textobjects.scm` found.
  - WARNING: No `folds.scm` query found for rust
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

## bash parser healthcheck
  - OK: bash parser found.
  - OK: `highlights.scm` found.
  - OK: `locals.scm` found.
  - WARNING: No `textobjects.scm` query found for bash
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `folds.scm` query found for bash
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

## ocaml_interface parser healthcheck
  - OK: ocaml_interface parser found.
  - OK: `highlights.scm` found.
  - OK: `locals.scm` found.
  - WARNING: No `textobjects.scm` query found for ocaml_interface
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - OK: `folds.scm` found.

## rst parser healthcheck
  - OK: rst parser found.
  - OK: `highlights.scm` found.
  - OK: `locals.scm` found.
  - OK: `textobjects.scm` found.
  - WARNING: No `folds.scm` query found for rst
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

## css parser healthcheck
  - OK: css parser found.
  - OK: `highlights.scm` found.
  - WARNING: No `locals.scm` query found for css
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `textobjects.scm` query found for css
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `folds.scm` query found for css
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

## c_sharp parser healthcheck
  - OK: c_sharp parser found.
  - OK: `highlights.scm` found.
  - WARNING: No `locals.scm` query found for c_sharp
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - OK: `textobjects.scm` found.
  - WARNING: No `folds.scm` query found for c_sharp
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

## javascript parser healthcheck
  - OK: javascript parser found.
  - OK: `highlights.scm` found.
  - OK: `locals.scm` found.
  - OK: `textobjects.scm` found.
  - OK: `folds.scm` found.

## php parser healthcheck
  - OK: php parser found.
  - OK: `highlights.scm` found.
  - OK: `locals.scm` found.
  - WARNING: No `textobjects.scm` query found for php
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - OK: `folds.scm` found.

## haskell parser healthcheck
  - OK: haskell parser found.
  - WARNING: No `highlights.scm` query found for haskell
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `locals.scm` query found for haskell
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `textobjects.scm` query found for haskell
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `folds.scm` query found for haskell
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

## tsx parser healthcheck
  - OK: tsx parser found.
  - OK: `highlights.scm` found.
  - OK: `locals.scm` found.
  - OK: `textobjects.scm` found.
  - OK: `folds.scm` found.
</code>
</details>

**Output of `nvim --version`**
```
NVIM v0.5.0-712-g4a996bc43
Build type: RelWithDebInfo
LuaJIT 2.1.0-beta3
Compilation: /usr/bin/cc -O2 -g -Og -g -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wshadow -Wconversion -Wmissing-prototypes -Wimplicit-fallthrough -Wvla -fstack-protector-strong -fno-common -fdiagnostics-color=always -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -DMIN_LOG_LEVEL=3 -I/home/pta2002/sources/neovim/build/config -I/home/pta2002/sources/neovim/src -I/home/pta2002/sources/neovim/.deps/usr/include -I/usr/include -I/home/pta2002/sources/neovim/build/src/nvim/auto -I/home/pta2002/sources/neovim/build/include
Compiled by pta2002@hydrogen

Features: +acl +iconv +tui
See ":help feature-compile"

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/local/share/nvim"

Run :checkhealth for more info
```

**Additional context**

**Comment by theHamsta** [comment]
For me this looks a lot like #526. The newest Python parser is too new for the current runtime in nvim.  We solved that problem by pinning the Python parser to a older version. `:TSUpdate python` should install a compatible version with the newest plugin revision.

**Comment by pta2002** [comment]
I tried doing that, but it still resulted in the same error

**Comment by theHamsta** [comment]
And you have the newest plugin version? Can you comment on how old/recent your nvim version is? I cannot find the commit hash shown in the version information. There's also a change in the highlighter interface #https://github.com/nvim-treesitter/nvim-treesitter/pull/540

**Comment by pta2002** [comment]
Yeah this is the latest version of nvim-treesitter. Neovim was compiled manually from sources, less than a week old.

**Comment by pta2002** [comment]
Manually installing an old version does work

**Comment by theHamsta** [comment]
> Manually installing an old version does work

I guess you might just have an old plugin revision without the `lockfile.json`. I am using the newest nvim revision and everything works for me.

**Comment by pta2002** [comment]
I'm on Linux, not sure how I would check this? But I do have both unzip and curl installed.

**Comment by theHamsta** [comment]
Does `:TSInstall python` look like  this for you
```
python parser already available: would you like to reinstall ? y/n: y
Downloading...
Extracting...
Compiling...
Treesitter parser for python has been installed
```

**Comment by pta2002** [comment]
Nope, doesn't say anything about extracting, just downloading.

**Comment by theHamsta** [comment]
Can you place a few `print` statements at https://github.com/nvim-treesitter/nvim-treesitter/blob/5debe931333b20fe2b352dd92c03ded837e3df4b/lua/nvim-treesitter/install.lua#L185 to see why the if branch is not taken?

**Comment by pta2002** [comment]
Hm if I do TSInstallSync it now says all the extracting and compiling stuff, but from what I could tell it didn't take that branch

**Comment by pta2002** [comment]
But now it isn't giving me any error anymore.

**Comment by theHamsta** [comment]
Fixed via #563
